src/components/data-exploration/embedding/Embedding.jsx

```diff
@@ -67,11 +70,16 @@ const Embedding = (props) => {
     hidden: cellSetHidden,
   } = cellSets;
 
+  // These are calculated to correctly position the popup according to the number of text rows in the popup
+  // There are 2 default rows: cellId and louvain clusters
+  const numCustomCellSets = cellSetHierarchy.children?.length || 0;
+  const numCellInfoTextRows = numCustomCellSets + 2;
```

Comment:
instead of doing this, can't you just get the height of CellInfo from within the CellInfo component? Similarly to invertX and invertY, this `numCellInfoTextRows` looks like something internal to CellInfo that CellInfo should worry about.

Reply:
Retrieved from `cellInfo`.